pennylane/devices/qutrit_mixed/measure.py

```diff
@@ -0,0 +1,309 @@
+            row_indices (str): Indices that must be replaced with sums
+            new_row_indices (str): Tensor indices of the state
+            **kwargs (dict): Stores indices calculated in `get_einsum_mapping`
+
+        Returns:
+            String of einsum indices to complete einsum calculations
+        """
+        op_1_indices = f"{new_row_indices}{row_indices}"
+
+        new_state_indices = get_new_state_einsum_indices(
+            old_indices=row_indices,
+            new_indices=new_row_indices,
+            state_indices=state_indices,
+        )
+
+        return f"{op_1_indices},...{state_indices}->...{new_state_indices}"
+
+    num_ch_wires = len(obs.wires)
+    einsum_indices = get_einsum_mapping(obs, state, map_indices, is_state_batched)
+    obs_mat = obs.matrix()
+    obs_shape = [QUDIT_DIM] * num_ch_wires * 2
+    obs_mat = math.cast(math.reshape(obs_mat, obs_shape), complex)
+    return math.einsum(einsum_indices, obs_mat, state)
+
+
+def trace_method(
+    measurementprocess: ExpectationMP, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Measure the expectation value of an observable by finding the trace of obs@rho.
+
+    Args:
+        measurementprocess (ExpectationMP): measurement process to apply to the state.
+        state (TensorLike): the state to measure.
+        is_state_batched (bool): whether the state is batched or not.
+
+    Returns:
+        TensorLike: the result of the measurement
+    """
+    obs = measurementprocess.obs
+    rho_mult_obs = apply_observable_einsum(obs, state, is_state_batched)
+
+    # using einsum since trace function axis selection parameter names
+    # are not consistent across interfaces, they don't exist for torch
+
+    num_wires = get_num_wires(state, is_state_batched)
+    trace = math.einsum(f"...{alphabet[:num_wires]*2}", rho_mult_obs)
+    return math.real(trace)
+
+
+def reduce_density_matrix(  # TODO: ask if I should have state diagonalization gates?
+    measurementprocess: StateMeasurement, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Get the state or reduced density matrix.
+
+    Args:
+        measurementprocess (StateMeasurement): measurement to apply to the state
+        state (TensorLike): state to apply the measurement to
+        is_state_batched (bool): whether the state is batched or not
+
+    Returns:
+        TensorLike: the resulting resquared density_matrix
+    """
+    wires = measurementprocess.wires
+    if not wires:
+        return resquare_state(state, get_num_wires(state, is_state_batched))
+
+    num_state_wires = get_num_wires(state, is_state_batched)
+    wires_to_trace = [x + is_state_batched for x in range(num_state_wires) if x not in wires]
+
+    for i, wire_to_trace in enumerate(wires_to_trace):
+        axis1 = wire_to_trace - i
+        axis2 = axis1 + num_state_wires - i
+
+        print(axis1, axis2)
+
+        state = math.trace(state, axis1=axis1, axis2=axis2)
```

Comment:
```suggestion
        state = math.trace(state, axis1=axis1, axis2=axis2)
```

Reply:
I just realized I should find a better way to do this too, it is dependent on axis1 and axis2 that will not work for all interfaces. Einsum should work well here though.

Reply:
I added my new method, hopefully this still looks good